### DISCUSSION TITLE
BigNumber comparison

### DISCUSSION THREAD

**Discussion by shmsula** [josdejong/mathjs#3039] [Q&A]
Tried to compare two big numbers (1000000000000), however only last version works.

![image](https://github.com/josdejong/mathjs/assets/133093698/4983943d-a3b9-4f11-a5bf-b39fd7ddebc3)

Is it correct behavior?

**Comment by josdejong** [comment]
The equality operators use an epsilon to compare numbers, which is `1e-12` by default. You will want to use a smaller epsilon.

See docs: https://mathjs.org/docs/datatypes/bignumbers.html#usage

**Comment by josdejong** [comment]
See also #2608, discussing the introduction of a separate epsilon for bignumbers to prevent the issue you're running into.
